src/Instrumentation/Curl/src/CurlInstrumentation.php

```diff
@@ -82,6 +82,14 @@ public static function register(): void
             pre: null,
             post: static function ($obj, array $params, mixed $retVal) use ($curlHandleToAttributes) {
                 if ($retVal != true) {
+                    if (curl_error($params[0])) {
+                        foreach ($params[1] as $option => $value) {
+                            if (!curl_setopt($params[0], $option, $value)) {
+                                break;
```

Comment:
If we `continue` here instead, would it permit setting the subsequent options too? (in case there is more than one error)

Reply:
yes it would, however I dont see a reason to do that since I think we should only be interested in recording options which would be used later (in the following curl handle execution).
trying to record all valid options might be useful (and could be done without `continue` too I think), but that is outside of scope of this PR.

Reply:
I think I agree with @vikmovcan here. We shouldn't set the options if curl itself wouldn't - so if `curl_setopt_array` wouldn't set them, neither should we.